### PR TITLE
feat(APIM-528): add API tests for upload file in deal folder endpoint

### PR DESCRIPTION
**Pull request by oscar-richardson-softwire**
## Introduction

When we completed [APIM-138](https://ukef-dtfs.atlassian.net/browse/APIM-138) and added the endpoint to upload files to SharePoint, we didn't include API or unit tests for this functionality.

## Resolution

This PR adds API (but not unit) tests for this functionality.

### CODE REVIEW

File: test/deal-folder/post-document-in-deal-folder.api-test.ts [hunk 1]
```diff
@@ -56,22 +67,327 @@ describe('postDocumentInDealFolder', () => {
...
+      minLength: 1,
+      maxLength: 250,
+      generateFieldValueOfLength: (length: number) => valueGenerator.buyerName({ length }),
+      pattern: BUYER_NAME.REGEX,
+      generateFieldValueThatDoesNotMatchRegex: () => ' Example Buyer',
+      validRequestBody: uploadFileInDealFolderRequest,
+      successStatusCode,
+      makeRequest: (requestBody: unknown[]) => makeRequestWithBodyAndMockReturningItemId(requestBody as UploadFileInDealFolderRequestDto),
+      givenAnyRequestBodyWouldSucceed: () => givenAnyRequestWouldSucceedUpToReturningItem(),
+    });
```
Comment (lmarrai-sw) [inline test/deal-folder/post-document-in-deal-folder.api-test.ts]: We test `buyerName` twice :)

**Review by lmarrai-sw — CHANGES_REQUESTED** [review]
Thanks @oscar-richardson-softwire

File: test/deal-folder/post-document-in-deal-folder.api-test.ts (hunk 1 above)
Comment (oscar-richardson-softwire) [reply test/deal-folder/post-document-in-deal-folder.api-test.ts]: Good spot, removed :)

File: src/constants/file-location-path.constant.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export const FILE_LOCATION_PATH = {
-  REGEX: /^[\w\-:/\\()\s]+$/,
+  // [a-fA-F\d]{24} matches a MongoDB hex, which will be the name of the first folder in the path.
+  REGEX: /^[a-fA-F\d]{24}(\/[\w\-:\\()\s]+)*$/,
```
Comment (lmarrai-sw) [inline src/constants/file-location-path.constant.ts:3]: It looks like we've accidentally changed the `REGEX` here 

EDIT: I was wrong, we didn't change the regex, but the linter is marking it as unsafe

**Review by lmarrai-sw — CHANGES_REQUESTED** [review]

File: src/constants/file-location-path.constant.ts (hunk 2 above)
Comment (lmarrai-sw) [reply src/constants/file-location-path.constant.ts:3]: The linter uses the [safe-regex](https://www.npmjs.com/package/safe-regex) package which can give false positives as it's a simple (but quick) check. I've put the regex into another checker (https://devina.io/redos-checker, which uses [recheck](https://github.com/makenowjust-labs/recheck)) which says that it's safe. I'll bring this up with the team to see if we want to move to a better but slower regex checker

**Review by lmarrai-sw — APPROVED** [review]

**Pull request merged**